### PR TITLE
Import defects

### PR DESCRIPTION
**Pull request by marioba**
Added "import defects" feature. 
It uses the external library openpyxl to read from Excel files.

### CODE REVIEW

File: modules/tools/importdefects.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                d = defects_module.LoadDefectsBase(self.iface, self.module_name)
+                defects_layers = d.run()
+                self.defects_list_dock.load_layers(defects_layers)
+
+            QApplication.restoreOverrideCursor()
+
+        except Exception:
+            QApplication.restoreOverrideCursor()
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            self.iface.messageBar().pushMessage("Error",
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: you can use raise VerisoErrorWithBar(.....)

File: modules/tools/importdefects.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    def init_gui(self):
+        return True
+
+    def accept(self):
+        QApplication.setOverrideCursor(Qt.WaitCursor)
+
+        xlsx = self.lineEditDefectsFile.text().strip()
+        load_checked = self.load_defects_check.isChecked()
+
+        header_list_punkte, rows_list_punkte = '', ''
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: header_list_points, rows_list_points. Let's try to use English only

File: modules/tools/importdefects.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        header_list_linien, rows_list_linien = '', ''
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: Same as above

File: modules/tools/importdefects.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        header_list_polygone, rows_list_polygone = '', ''
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: Same as above

File: modules/tools/importdefects.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            return
+
+        self.wb = load_workbook(filename = xlsx, read_only = True)
+
+        try:
+            header_list_punkte, rows_list_punkte = self.read_sheet(u'Mängelliste (Punkte)')
+            header_list_linien, rows_list_linien = self.read_sheet(u'Mängelliste (Linien)')
+            header_list_polygone, rows_list_polygone = self.read_sheet(u'Mängelliste (Polygone)')
+        except WrongExcelError as e:
+            self.iface.messageBar().pushMessage('Wrong Excel file', e.value,
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: VerisoErrorWithBar or if you want  QgsMessageBar::pushCritical

File: modules/tools/importdefects.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+            return
+
+        punkte_query = self.create_query('t_maengel_punkt', header_list_punkte, rows_list_punkte)
+        linien_query = self.create_query('t_maengel_linie', header_list_linien, rows_list_linien)
+        polygone_query = self.create_query('t_maengel_polygon', header_list_polygone, rows_list_polygone)
+
+        try:
+            self.open_db()
+
+            self.execute_query(punkte_query)
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: english

File: modules/tools/importdefects.py [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            self.execute_query(linien_query)
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: english

File: modules/tools/importdefects.py [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+            self.execute_query(polygone_query)
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: english

File: modules/tools/importdefects.py [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+    def on_btnBrowseDefectsFile_clicked(self):
+        file_path = QFileDialog.getOpenFileName(
+                self,
+                tr("Choose defects file"),
+                self.input_xlsx_path,
+                "XLSX (*.xlsx)")
+        file_info = QFileInfo(file_path)
+        self.lineEditDefectsFile.setText(file_info.absoluteFilePath())
+
+class QueryExecutionError(Exception):
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: move to base/utils/exceptions.py

File: modules/tools/importdefects.py [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+    def __init__(self, value):
+        self.value = value
+        def __str__(self):
+            return repr(self.value)
+
+class WrongExcelError(Exception):
```
Comment (mbernasocchi) [inline modules/tools/importdefects.py]: move to base/utils/exceptions.py

**Comment by marioba** [comment]
ok, hai fatto bene. appena riesco le sistemo.

**Pull request merged**